src/cli/commands/add.js

```diff
@@ -120,7 +128,7 @@ export class Add extends Install {
    */
 
   async init(): Promise<Array<string>> {
-    if (this.config.workspaceRootFolder && this.config.cwd === this.config.workspaceRootFolder) {
+    if (this.ignoreWorkspaces) {
```

Comment:
Unless I'm mistaken, this condition cannot be replaced this way; it is meant to ensure that running `yarn add <something>` inside the workspace root will warn the user that they probably don't want to do this. With this change, no warning will be emitted in such a case because `ignoreWorkspaces` will be `false` in traditional `add` settings. I think a better condition is:

```js
this.config.workspaceRootFolder && this.config.cwd === this.config.workspaceRootFolder && !this.ignoreWorkspaces
```

Reply:
That conditional wouldn't ever be true though according to this logic right? https://github.com/kaylieEB/yarn/blob/93b2aaeb33aad23218177f3a1ed2bd9870e4d077/src/cli/commands/add.js#L27-L29. I've also tested that it still warns when I try to add a non-dev dependency.